demos/portal-embed-demo/public/js/ttt/ttt-reco.js

```diff
@@ -0,0 +1,169 @@
+            }
+            .thumbnail {
+                height: 100%;
+                float:left;
+            }
+            .thumbnail img{
+                height: 100%;
+            }
+            .description {
+                float:left;
+                margin:20px 0px 10px 20px;
+            }
+            .title{
+                font-size: 1.0em;
+            }
+            .series{
+                font-size: 0.8em;
+                margin-top: 5px;
+                font-weight: 700;
+            }
+            .add {
+                height: 100%;
+                float:right;
+                border-left: 1px solid #DDD;
+                width: 55px;
+                text-align: center;
+                position: relative;
+            }
+            .add-button {
+                width: 100%;
+                height: 100%;
+                position: absolute;
+                top: 0px;
+                left: 0px;
+            }
+            .add img{
+                height: 50%;
+                margin-top:20px;
+                margin-left:5px;
+            }
+            .ripple {
+                background-position: center;
+                transition: background 0.8s;
+            }
+            .ripple:hover {
+                background: #FBFBFB radial-gradient(circle, 
+                    transparent 2%, #FBFBFB 1%) center/15000%;
+            }
+            .ripple:active {
+                background-color: #DDD;
+                background-size: 100%;
+                transition: background 0s;
+            }
+        `;
+    }
+
+    /**
+     * HTML template
+     */
+    get template() {
+        return `
+                <div class='header web-font'>Listen more of TTT</div>
+                ${this.recommendationList.map(key => {
+                    const imgURL = audioList[key].imgURL;
+                    const title = audioList[key].title;
+                    return this._createItemTemplate(key, imgURL, title);
+                }).join('')}
+        `
+    }
+
+    /**
+     * Generating HTML for each recommended auido
+     * @param {String} key 
+     * @param {String} imgURL 
+     * @param {String} title 
+     * @return {String}
+     */
+    _createItemTemplate(key, imgURL, title) {
+        return `
+            <div class='pod'>
+                <div class='thumbnail'>
+                    <img src=${imgURL}>
+                </div>
+                <div class='description'>
+                    <div class='title'>${title}</div>
+                    <div class='series'>Tottaly Tooling Tips</div>
+                </div>
+                <div class='add ripple'>
+                    <img src='/img/add.png'>
+                    <div class='add-button' added='false' id=${key}></div>
+                </div>
+            </div>
+        `;
+    }
+
+    /**
+     * Adding event listeners for the buttons
+     */
+    _hookEvents() {
+        [...this.root.querySelectorAll('.add-button')].map(elm => {
```

Comment:
We don't use the result of this map, so it would be more idiomatic to just write:

```javascript
for (let elm of this.root.querySelectorAll('.add-button')) {
    /* ... */
}
```

Reply:
Fixed it!